test/test_c10d.py

```diff
@@ -2177,30 +2185,39 @@ def test_find_unused_parameters(find_unused_parameters, test_default=False):
                     FindUnusedParametersModule().float().to(device_id),
                     device_ids=[device_id],
                     process_group=process_group,
+                    delay_allreduce=self.delay_allreduce,
                 )
             else:
                 model = DistributedDataParallel(
                     FindUnusedParametersModule().float().to(device_id),
                     device_ids=[device_id],
                     process_group=process_group,
                     find_unused_parameters=find_unused_parameters,
+                    delay_allreduce=self.delay_allreduce,
                 )
 
             output, fc3 = model(input)
             output = fc3(output)
             loss = criterion(output, target)
             loss.backward()
 
-        # First test that finding unused params under these conditions is to
-        # trigger an error when `backward` is called (because fc3 is an unused
-        # parameter and will therefore be marked ready twice).
-        try:
-            test_find_unused_parameters(True)
-        except Exception as ex:
-            self.assertTrue(
-                str(ex).startswith("Expected to mark a variable ready only once."))
-        else:
-            self.fail("Expected exception")
+        if not self.delay_allreduce:
+            # First test that finding unused params under these conditions is to
+            # trigger an error when `backward` is called (because fc3 is an unused
+            # parameter and will therefore be marked ready twice).
+            #
+            # However, this should work if `delay_allreduce` is set to true,
```

Comment:
Where do we verify that `test_find_unused_parameters` returns successfully under when `delay_allreduce` is True?